driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -151,10 +151,46 @@ public void onSuccess(List<Void> l) {
                     initFuture.set(null);
                 }
             }
+
+            @Override
+            public void onFailure(Throwable t) {
```

Comment:
Isn't this at odds with JAVA-544 (#420) since if any connections fail onFailure will be invoked which will close all connections that may have succeeded, mark this future as failed which will cause the pool to not be set in SessionManager.pools in SessionManager.forceRenewPool?